src/sentry/explore/endpoints/explore_saved_queries.py

```diff
@@ -300,27 +300,24 @@ def get(self, request: Request, organization) -> Response:
         if not self.has_feature(organization, request):
             return self.respond(status=404)
 
-        if features.has(
-            "organizations:performance-default-explore-queries", organization, actor=request.user
-        ):
-            try:
-                lock = locks.get(
-                    f"explore:sync_prebuilt_queries:{organization.id}:{request.user.id}",
-                    duration=10,
-                    name="sync_prebuilt_queries",
-                )
-                with lock.acquire():
-                    # Adds prebuilt queries to the database if they don't exist.
-                    # Updates them if they are outdated.
-                    # Deletes old prebuilt queries from the database if they should no longer exist.
-                    # Stars prebuilt queries for the user if it is the first time they are being fetched by the user.
-                    sync_prebuilt_queries(organization)
-                    sync_prebuilt_queries_starred(organization, request.user.id)
-            except UnableToAcquireLock:
-                # Another process is already syncing the prebuilt queries. We can skip syncing this time.
-                pass
-            except Exception as err:
-                sentry_sdk.capture_exception(err)
+        try:
+            lock = locks.get(
```

Comment:
I'm also curious why we didn't use atomic transactions here?

Reply:
Atomic transactions ensure all db queries within the transaction either succeed together, or they all fail. They don't prevent concurrent calls from executing, which is what this lock solves

Reply:
Gotcha yeah I forgot haha.

Not that it's a problem, more from understanding.
This looks like an application lock and not a database lock? If so, wouldn't a lock not achieve the desired result if we had multiple instances of sentry running.

Reply:
Yes if you have multiple instances running and you send concurrent requests that hit different instances, you could potentially get around this lock. I'm not very familiar with what kind of load balancing or routing we have in place, but from my testing this doesn't seem to be causing issues yet. Also, this lock isn't fully necessary for the prebuilt queries feature to work, it just prevents extra db calls for this edge case. At worst, the db transaction should just error due to constraints which gets caught and passed anyways.